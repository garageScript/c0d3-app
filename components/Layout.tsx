import React from 'react'
import AppNav from './AppNav'
import Footer from './Footer'
import Head from 'next/head'
const Layout: React.FC<{ title?: string }> = ({ children, title }) => (
  <>
    <Head>
      <title key="title">{title ? `${title} â€” C0D3` : 'C0D3'}</title>
    </Head>
    <input type="text"></input>
    <AppNav />
    <div className="container-md">{children}</div>
    <Footer />
  </>
)

export default Layout
