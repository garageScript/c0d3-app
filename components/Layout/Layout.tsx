import React from 'react'
import AppNav from '../AppNav'
import Footer from '../Footer'
import Head from 'next/head'
import type { LayoutGetter } from '../../@types/page'
import styles from './layout.module.scss'

const Layout: React.FC<{ title?: string }> = ({ children, title }) => (
  <>
    <Head>
      <title key="title">{title ? `${title} â€” C0D3` : 'C0D3'}</title>
    </Head>
    <AppNav />
    <div className={`container-md ${styles.container}`}>{children}</div>
    <Footer />
  </>
)

export const getLayout: LayoutGetter = page => <Layout>{page}</Layout>
export default Layout
