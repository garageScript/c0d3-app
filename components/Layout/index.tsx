export { getLayout, default } from './Layout'
