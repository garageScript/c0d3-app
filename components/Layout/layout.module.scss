.container {
  margin-top: 32px;
}
