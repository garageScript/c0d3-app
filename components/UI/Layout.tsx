import React from 'react'

const Layout: React.FC = ({ children }) => {
  return (
    <div className="container">
      {children}
      <style jsx>{`
        .container {
          display: grid;
          grid-template-columns: repeat(12, 1fr);
          grid-gap: 1em;
          max-width: 1440px;
          margin: auto;
          text-align: center;
          margin-bottom: 1em;
        }

        /* 
         * We can pre setup default column classes like below
         * Personally I prefear write this css into my components
         */
        .col-1 {
          grid-column: span 1;
          background-color: #e2e8f0;
        }
        .col-2 {
          grid-column: span 2;
          background-color: #e2e8f0;
        }
        .col-3 {
          grid-column: span 3;
          background-color: #e2e8f0;
        }
        .col-4 {
          grid-column: span 4;
          background-color: #e2e8f0;
        }
        .col-5 {
          grid-column: span 5;
          background-color: #e2e8f0;
        }
        .col-6 {
          grid-column: span 6;
          background-color: #e2e8f0;
        }
        .col-7 {
          grid-column: span 7;
          background-color: #e2e8f0;
        }
        .col-8 {
          grid-column: span 8;
          background-color: #e2e8f0;
        }
        .col-9 {
          grid-column: span 9;
          background-color: #e2e8f0;
        }
        .col-10 {
          grid-column: span 10;
          background-color: #e2e8f0;
        }
        .col-11 {
          grid-column: span 11;
          background-color: #e2e8f0;
        }
        .col-12 {
          grid-column: span 12;
          background-color: #e2e8f0;
        }
      `}</style>
    </div>
  )
}

export default Layout
