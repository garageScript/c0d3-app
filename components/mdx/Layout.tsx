import React from 'react'
import Layout from '../Layout'

const LessonLayout: React.FC = ({ children }) => {
  return (
    <Layout>
      <div className={`card shadow-sm mt-3 d-block border-0 p-4 bg-white`}>
        {children}
      </div>
    </Layout>
  )
}
export default LessonLayout
