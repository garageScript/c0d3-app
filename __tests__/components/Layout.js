import React from 'react'
import { render } from '@testing-library/react'
import Layout from '../../components/Layout'
import SessionContext from '../../helpers/contexts/session'

describe('Layout Component', () => {
  test('Should render correctly when no user in session', () => {
    const session = { data: null }
    const tree = (
      <SessionContext.Provider value={session}>
        <Layout />
      </SessionContext.Provider>
    )

    const { container } = render(tree)
    expect(container).toMatchSnapshot()
  })

  test('Should render correctly when user session found', () => {
    const session = {
      data: { userInfo: { name: 'tester', username: 'tester' } }
    }
    const tree = (
      <SessionContext.Provider value={session}>
        <Layout />
      </SessionContext.Provider>
    )

    const { container } = render(tree)
    expect(container).toMatchSnapshot()
  })

  test('Should render correct with no session context', () => {
    const { container } = render(<Layout />)
    expect(container).toMatchSnapshot()
  })
})
