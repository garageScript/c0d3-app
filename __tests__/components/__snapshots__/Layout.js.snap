// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Should render correct with no session context 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand text-primary font-weight-bold"
        href="/"
      >
        C0D3
      </a>
      <div
        id="navbarNav"
      >
        <div
          class="navbar-nav collapse navbar-collapse"
        >
          <div
            class="navbar-nav collapse navbar-collapse"
          >
            <a
              class="nav-item nav-link active"
              href="/"
            >
              Home 
              <span
                class="sr-only"
              >
                (current)
              </span>
            </a>
            <a
              class="nav-item nav-link"
              href="/#learning"
            >
              Learning Process
            </a>
            <a
              class="nav-item nav-link"
              href="https://c0d3.com/book"
              rel="noopener noreferrer"
              target="_blank"
            >
              Resources
            </a>
            <a
              class="nav-item nav-link"
              href="https://chat.c0d3.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Help
            </a>
          </div>
        </div>
      </div>
      <div>
        <a
          class="btn btn-secondary border mr-3"
          href="/login"
        >
          Login
        </a>
        <a
          class="btn btn-secondary border mr-3"
          href="/signup"
        >
          Signup
        </a>
      </div>
    </div>
  </nav>
  <div
    class="container"
  />
  <footer
    class="mt-5 font-weight-light text-center pb-5 text-muted"
  >
    © Copyright 2020 GarageScript
    <br />
    All Rights Reserved
  </footer>
</div>
`;

exports[`Layout Component Should render correctly when no user in session 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand text-primary font-weight-bold"
        href="/"
      >
        C0D3
      </a>
      <div
        id="navbarNav"
      >
        <div
          class="navbar-nav collapse navbar-collapse"
        >
          <div
            class="navbar-nav collapse navbar-collapse"
          >
            <a
              class="nav-item nav-link active"
              href="/"
            >
              Home 
              <span
                class="sr-only"
              >
                (current)
              </span>
            </a>
            <a
              class="nav-item nav-link"
              href="/#learning"
            >
              Learning Process
            </a>
            <a
              class="nav-item nav-link"
              href="https://c0d3.com/book"
              rel="noopener noreferrer"
              target="_blank"
            >
              Resources
            </a>
            <a
              class="nav-item nav-link"
              href="https://chat.c0d3.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Help
            </a>
          </div>
        </div>
      </div>
      <div>
        <a
          class="btn btn-secondary border mr-3"
          href="/login"
        >
          Login
        </a>
        <a
          class="btn btn-secondary border mr-3"
          href="/signup"
        >
          Signup
        </a>
      </div>
    </div>
  </nav>
  <div
    class="container"
  />
  <footer
    class="mt-5 font-weight-light text-center pb-5 text-muted"
  >
    © Copyright 2020 GarageScript
    <br />
    All Rights Reserved
  </footer>
</div>
`;

exports[`Layout Component Should render correctly when user session found 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand text-primary font-weight-bold"
        href="/"
      >
        C0D3
      </a>
      <div
        id="navbarNav"
      >
        <div
          class="navbar-nav collapse navbar-collapse"
        >
          <div
            class="navbar-nav collapse navbar-collapse"
          >
            <a
              class="nav-item nav-link active"
              href="/curriculum"
            >
              Curriculum 
              <span
                class="sr-only"
              >
                (current)
              </span>
            </a>
            <a
              class="nav-item nav-link"
              href="#"
            >
              Repo
            </a>
            <a
              class="nav-item nav-link"
              href="#"
            >
              Journey
            </a>
            <a
              class="nav-item nav-link"
              href="https://chat.c0d3.com"
            >
              Help
            </a>
          </div>
        </div>
      </div>
      <div>
        <button
          class="btn border btn-secondary overflow-hidden p-2 text-truncate"
        >
          tester
        </button>
        <button
          class="btn border btn-secondary ml-2"
        >
          Logout
        </button>
      </div>
    </div>
  </nav>
  <div
    class="container"
  />
  <footer
    class="mt-5 font-weight-light text-center pb-5 text-muted"
  >
    © Copyright 2020 GarageScript
    <br />
    All Rights Reserved
  </footer>
</div>
`;
